community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingNeoStores.java

```diff
@@ -77,6 +80,8 @@
  */
 public class BatchingNeoStores implements AutoCloseable, MemoryStatsVisitor.Visitable
 {
+    // Empirical and slightly defensive threshold where relationship records seem to start requiring double record units.
+    private static final long DOUBLE_RELATIONSHIP_RECORD_UNIT_THRESHOLD = 1L << 33;
```

Comment:
while this empirical metric can be good enough, i think we need to add a bit of description how it was built (based on what assumptions) or direct future us into "requiredDataLength" from record formats etc.